File: client/src/data/waitlist.ts
export interface Waitlist {
  id: string;
  firstname: string;
  lastname: string;
  email: string;
  phone: string;
  preferences: string;
}

export const mockWaitlist: Waitlist[] = [
  {
    id: "1",
    firstname: "Customer",
    lastname: "1",
    email: "customer@aprilslilpugs.com",
    phone: "(662) 555-0199",
    preferences: "I want a black male",
  },
  {
    id: "2",
    firstname: "Customer",
    lastname: "2",
    email: "customer@aprilslilpugs.com",
    phone: "(662) 555-0199",
    preferences: "I want a red female",
  },
  {
    id: "3",
    firstname: "Customer",
    lastname: "3",
    email: "customer@aprilslilpugs.com",
    phone: "(662) 555-0199",
    preferences: "I want a fawn male",
  },
];

File: server/internal/models/waitlist.go
package models

import "time"

type Waitlist struct {
	ID          int       `json:"id"`
	FirstName   string    `json:"firstname" form:"firstname" binding:"required"`
	LastName    string    `json:"lastname" form:"lastname" binding:"required"`
	Email       string    `json:"email" form:"email" binding:"required,email"`
	Phone       string    `json:"phone" form:"phone"`
	Preferences string    `json:"preferences" form:"preferences"`
	Status      string    `json:"status" form:"status"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}
